### ISSUE TITLE
[Toolbar Redesign] Toolbar’s Share menu copies the current page’s title instead of its URL

### ISSUE THREAD

**Issue by cpeterso** [mozilla-mobile/firefox-ios#23346]
## Steps to reproduce

1. Open a web page, such as https://www.wikipedia.org/
2. Using the new toolbar’s Share button, copy the URL. 
2. Switch to another app.
3. Paste the URL into other app.

### Expected behavior

The page’s URL https://www.wikipedia.org/ should be pasted. 

### Actual behavior

The page’s title “Wikipedia” is pasted. 

### Device & build information

* Device: iPhone 15 Plus running iOS 18.1.1
* Build version: Nightly 9000 (47935)
* First seen version: This bug was a regression in Nightly 9000 (47686). The Share button copies the URL correctly in the previous build, Nightly 9000 (47636).

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-10672)

**Comment by cpeterso** [comment]
This Share menu bug might be related to bookmark sharing bug https://github.com/mozilla-mobile/firefox-ios/issues/23339

**Comment by ih-codes** [comment]
Hi @cpeterso, thanks for flagging this. This was a regression caused by some work done on the Info Card Referral experiment (I stepped in to help on the iOS side when I noticed there were some bugs). I merged a fix for this on Friday. I just checked on Nightly 48005, and this should be resolved. I'm going to mark this ticket as completed / duplicate of #23339 unless you are still seeing issues after updating. Thanks!
